src/certificate.rs

```diff
@@ -0,0 +1,21 @@
+use std::fmt;
+
+#[derive(Clone)]
+pub struct Certificate {
+    pub inner_cert: native_tls::Certificate,
+}
+
+impl fmt::Debug for Certificate {
```

Comment:
I am not sure if the debug output is worth the additional module and new type wrapper compared to e.g. just using the existing `SkipDebug` wrapper? Especially so since we start to expose this in our API as per the base settings member.

If we decide to keep it, I would suggest using a tuple type like `pub struct Certificate(pub native_tls::Certificate)` and dropping the `From` impl as we never need to convert this in generic context.

Reply:
> Especially so since we start to expose this in our API as per the base settings member.

It's not exposed. The public `add_root_certificate` methods accepts a `native_tls::Certificate`.

Reply:
> It's not exposed. The public add_root_certificate methods accepts a native_tls::Certificate.

Indeed. I was mistakenly thinking about the `pub root_certificates: Vec<Certificate>` field in `BaseSettings` but that type itself is not exposed in the API.

Personally, I would still prefer to just use `SkipDebug` there to keep this reasonably simple.

Reply:
Sure. I just didn't knew there was a `SkipDebug` trait in the first place. I've just amended the PR so it uses it instead of the new `Certificate` module/type.

Reply:
Looks much nicer to me now! (Totally a nit, but I guess `SkipDebug<Vec<Certificate>>` would be nicer as it would avoid having to spell every time an item is added and the resulting output will contain only one ellipsis instead of one for each certificate. `SkipDebug` also has a similar `From` impl as `Certificate` had.)

Reply:
Indeed, that's better this way. PR updated.